spring-boot-project/spring-boot/src/main/java/org/springframework/boot/SpringApplication.java

```diff
@@ -558,16 +558,6 @@ protected void configurePropertySources(ConfigurableEnvironment environment, Str
 	protected void configureProfiles(ConfigurableEnvironment environment, String[] args) {
 	}
 
-	private void configureAdditionalProfiles(ConfigurableEnvironment environment) {
```

Comment:
The additional profiles are included in the Environment's active profiles at ConfigDataEnvironmentPostProcessor when legacy processing is used, or at ConfigDataEnvironment as below
https://github.com/spring-projects/spring-boot/blob/4ad87a861d34174e9ea7209304985eaba929d738/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/config/ConfigDataEnvironment.java#L348

Therefore, `configureAdditionalProfiles` become unnecessary.